sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1774,6 +1775,35 @@ class Analyzer(
   }
 
   /**
+   * Substitute Hints.
+   * - BROADCAST/BROADCASTJOIN/MAPJOIN match the closest table with the given name parameters.
```

Comment:
It sounds like you remove the comment you put before. 

Second, we also need to explain the restriction and the future work.

Third, we also need to document the issue you hit in the PR description. It can help us understand the issues when we enhance it later.